resolwe/flow/executors/docker/__init__.py

```diff
@@ -30,15 +35,56 @@ def start(self):
             'container_image': self.requirements.get('image', settings.FLOW_EXECUTOR['CONTAINER_IMAGE']),
         }
 
+        # Set resource limits.
+        limits = []
+        if 'cores' in self.resources:
+            # Each core is equivalent to 1024 CPU shares. The default for Docker containers
+            # is 1024 shares (we don't need to explicitly set that).
+            limits.append('--cpu-shares={}'.format(int(self.resources['cores']) * 1024))
+
+        # By default, we set the memory limit to 4GiB.
+        memory = int(self.resources.get('memory', 4096))
+        # Set both memory and swap limits as we want to enforce the total amount of memory
+        # used (otherwise swap would not count against the limit).
+        limits.append('--memory={0}m --memory-swap={0}m'.format(memory))
+
+        command_args['limits'] = ' '.join(limits)
+
         # set container name
         container_name_prefix = getattr(settings, 'FLOW_EXECUTOR', {}).get('CONTAINER_NAME_PREFIX', 'resolwe')
         command_args['container_name'] = '--name={}_{}'.format(container_name_prefix, self.data_id)
 
-        # Configure Docker network mode for the container (if specified).
-        # By default, current Docker versions use the 'bridge' mode which
-        # creates a network stack on the default Docker bridge.
-        network = getattr(settings, 'FLOW_EXECUTOR', {}).get('NETWORK', '')
-        command_args['network'] = '--net={}'.format(network) if network else ''
+        if 'network' in self.resources:
+            # Configure Docker network mode for the container (if specified).
+            # By default, current Docker versions use the 'bridge' mode which
+            # creates a network stack on the default Docker bridge.
+            network = getattr(settings, 'FLOW_EXECUTOR', {}).get('NETWORK', '')
+            command_args['network'] = '--net={}'.format(network) if network else ''
+        else:
+            # No network if not specified.
+            command_args['network'] = '--net=none'
+
+        # Security options.
+        security = []
+
+        # Generate and set seccomp policy to limit syscalls.
+        self.policy_file = tempfile.NamedTemporaryFile(mode='w')
+        json.dump(SECCOMP_POLICY, self.policy_file)
+        self.policy_file.file.flush()
+        if not strtobool(getattr(settings, 'RESOLWE_DOCKER_DISABLE_SECCOMP', '0')):
```

Comment:
I think it would be better to move `strtobool` to settings and also set default there.